### ISSUE TITLE
PushFlows ignores the result of PushCommand.call() with BitBucket

### ISSUE THREAD

**Issue by lburja** [kestra-io/plugin-git#89]
### Describe the issue

I have a situation where the `PushFlows` task finishes successfully, but I see no commits in the remote git repo (BitBucket).

Here is a screenshot from Kestra, where the log output claims that changes were pushed:
<img width="1027" alt="image" src="https://github.com/user-attachments/assets/fb4cef02-f46a-4a03-99eb-6b0e2480b351">

However, when I check the remote repository, no commits have been made in fact.

By looking in the source code, I see that the result of `PushCommand.call()` is ignored:
![image](https://github.com/user-attachments/assets/e492b7a4-d36f-4263-9ee2-db7d1c1da63e)

According to this StackOverflow question, not all push errors result in an exception being thrown, some of them are returned as push results : https://stackoverflow.com/questions/51140394/jgit-push-wont-fail-when-push-is-impossible
I believe that the push fails in some way, but there's no way to see the error in Kestra.

I would expect at the very least that any push result is logged. Even better, it would be returned as an output, and the task would throw an exception if any update was rejected.

### Environment

- Kestra Version: v0.18.2
- Operating System (OS/Docker/Kubernetes): Kubernetes
- Java Version (if you don't run kestra in Docker): n/a

**Comment by lburja** [comment]
@smunteankestra The point is that PushCommand::call() returns a value (`Iterable<PushResult>`), which might contain status OK, or status REJECTED_OTHER_REASON, or ....

![image](https://github.com/user-attachments/assets/6d201f00-ff19-4ce0-b132-13ccf272fdf4)

But, Kestra ignores the result (see `AbstractPushTask`):

![image](https://github.com/user-attachments/assets/1c5b6cbe-7e26-400a-aeeb-46b2920bf09e)

**Comment by anna-geller** [comment]
great, thx so much for thorough examination @smunteankestra 💯 

so next steps: **add error handling for when push doesn't work when commit signature verification is not enabled**

**Comment by ramiz-rahman** [comment]
Hi @anna-geller, I'm running into a similar issue where Kestra is telling me that the commit is made but there are no changes in my GitHub Repo. If you click on the output commit link, it takes you to a 404 page. 

If the `push_flows` task failed with an helpful error message, I could debug further, but right now, I'm at a loss. 

The `push_namespace_files` task actually worked as expected and committed changes to the repo. This proves that the GH_Token is correct and has the right permissions. 

Github Repo URL: https://github.com/ramiz-rahman/de-zoomcamp-2025/tree/kestra

I've spent a good few hours trying to solve this on my own before searching up and commenting on this issue. 

Flow:
```
id: push-to-git
namespace: system

variables:
  gh_username: ramiz-rahman
  gh_repo: de-zoomcamp-2025
  gh_repo_url: https://github.com/ramiz-rahman/de-zoomcamp-2025.git

tasks:
  - id: push_flows
    type: io.kestra.plugin.git.PushFlows
    sourceNamespace: zoomcamp
    gitDirectory: 02_week_kestra/_flows
    url: "{{ vars.gh_repo_url }}"
    username: "{{ vars.gh_username }}"
    password: "{{ kv('GH_TOKEN') }}"
    branch: kestra
    commitMessage: add flows {{ now() }}
    dryRun: false

  - id: push_namespace_files
    type: io.kestra.plugin.git.PushNamespaceFiles
    namespace: zoomcamp
    gitDirectory: 02_week_kestra/_files
    url: "{{ vars.gh_repo_url }}"
    username: "{{ vars.gh_username }}"
    password: "{{ kv('GH_TOKEN') }}"
    branch: kestra
    commitMessage: add namespace files {{ now() }}
    dryRun: false

  # - id: create_pr
  #   type: io.kestra.plugin.github.pulls.Create
  #   oauthToken: "{{ kv('GH_TOKEN') }}"
  #   repository: "{{ vars.gh_username }}/{{ vars.gh_repo }}"
  #   sourceBranch: kestra
  #   targetBranch: main
  #   title: Merge kestra to main 
  #   body: "Request to merge changes from kestra to main"

# triggers:
#   - id: every_full_hour
#     type: io.kestra.plugin.core.trigger.Schedule
#     cron: "*/15 * * * *"
```

Kestra Log for push flows
```
2025-01-28 11:10:42.727Start cloning from 'https://github.com/ramiz-rahman/de-zoomcamp-2025.git'
2025-01-28 11:10:44.347Pushing to https://github.com/ramiz-rahman/de-zoomcamp-2025.git on branch kestra
2025-01-28 11:10:45.950Pushed to https://github.com/ramiz-rahman/de-zoomcamp-2025.git/commit/bb3449d8b5cdda08738d2101cba8a65df0b08a83
2025-01-28 11:10:45.973{changes:"0",file:"02_week_kestra/_flows/02_postgres_taxi.yml",deletions:"-0",additions:"+270"}
2025-01-28 11:10:45.975{changes:"0",file:"02_week_kestra/_flows/02_postgres_taxi_scheduled.yml",deletions:"-0",additions:"+275"}
2025-01-28 11:10:45.977{changes:"0",file:"02_week_kestra/_flows/03_postgres_dbt.yml",deletions:"-0",additions:"+60"}
2025-01-28 11:10:45.979{changes:"0",file:"02_week_kestra/_flows/04_gcp_kv.yml",deletions:"-0",additions:"+46"}
2025-01-28 11:10:45.981{changes:"0",file:"02_week_kestra/_flows/05_gcp_setup.yml",deletions:"-0",additions:"+22"}
2025-01-28 11:10:45.982{changes:"0",file:"02_week_kestra/_flows/06_gcp_taxi.yml",deletions:"-0",additions:"+248"}
2025-01-28 11:10:45.984{changes:"0",file:"02_week_kestra/_flows/06_gcp_taxi_scheduled.yml",deletions:"-0",additions:"+249"}
```

Kestra Log for 
```
2025-01-28 11:10:46.811Start cloning from 'https://github.com/ramiz-rahman/de-zoomcamp-2025.git'
2025-01-28 11:10:48.369Pushing to https://github.com/ramiz-rahman/de-zoomcamp-2025.git on branch kestra
2025-01-28 11:10:48.375No changes to commit. Skipping push.
2025-01-28 11:10:48.401{changes:"0",file:".DS_Store",deletions:"-0",additions:"+0"}
2025-01-28 11:10:48.402{changes:"0",file:".gitignore",deletions:"-0",additions:"+40"}
2025-01-28 11:10:48.403{changes:"0",file:"01_week_docker_sql_terraform/docker-compose.yaml",deletions:"-0",additions:"+48"}
2025-01-28 11:10:48.403{changes:"0",file:"01_week_docker_sql_terraform/README.md",deletions:"-0",additions:"+285"}
2025-01-28 11:10:48.404{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/.gitignore",deletions:"-0",additions:"+117"}
2025-01-28 11:10:48.405{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/Dockerfile",deletions:"-0",additions:"+20"}
2025-01-28 11:10:48.406{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/README.md",deletions:"-0",additions:"+12"}
2025-01-28 11:10:48.407{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/environment.yml",deletions:"-0",additions:"+17"}
2025-01-28 11:10:48.408{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/notebooks/pipeline.ipynb",deletions:"-0",additions:"+1087"}
2025-01-28 11:10:48.408{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/__init__.py",deletions:"-0",additions:"+0"}
2025-01-28 11:10:48.409{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/extract.py",deletions:"-0",additions:"+63"}
2025-01-28 11:10:48.410{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/load.py",deletions:"-0",additions:"+47"}
2025-01-28 11:10:48.410{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/logging_config.py",deletions:"-0",additions:"+41"}
2025-01-28 11:10:48.411{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/pipeline.py",deletions:"-0",additions:"+88"}
2025-01-28 11:10:48.412{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/transformers.py",deletions:"-0",additions:"+28"}
2025-01-28 11:10:48.413{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/utils/__init__.py",deletions:"-0",additions:"+3"}
2025-01-28 11:10:48.414{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/utils/download.py",deletions:"-0",additions:"+88"}
2025-01-28 11:10:48.414{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/utils/database.py",deletions:"-0",additions:"+50"}
2025-01-28 11:10:48.415{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/script.py",deletions:"-0",additions:"+81"}
2025-01-28 11:10:48.415{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/ny_taxi_etl_pipeline/utils/logging_utils.py",deletions:"-0",additions:"+32"}
2025-01-28 11:10:48.416{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/tests/test_transform.py",deletions:"-0",additions:"+55"}
2025-01-28 11:10:48.416{changes:"0",file:"01_week_docker_sql_terraform/ny_taxi_etl_pipeline/tests/__init__.py",deletions:"-0",additions:"+0"}
2025-01-28 11:10:48.417{changes:"0",file:"01_week_docker_sql_terraform/terraform/main.tf",deletions:"-0",additions:"+25"}
2025-01-28 11:10:48.417{changes:"0",file:"01_week_docker_sql_terraform/terraform/.terraform.lock.hcl",deletions:"-0",additions:"+22"}
2025-01-28 11:10:48.418{changes:"0",file:"02_week_kestra/_files/README.md",deletions:"-0",additions:"+38"}
2025-01-28 11:10:48.418{changes:"0",file:"01_week_docker_sql_terraform/terraform/variables.tf",deletions:"-0",additions:"+34"}
2025-01-28 11:10:48.419{changes:"0",file:"02_week_kestra/_files/analyses/hack-load-data.sql",deletions:"-0",additions:"+49"}
2025-01-28 11:10:48.419{changes:"0",file:"02_week_kestra/_files/dbt_project.yml",deletions:"-0",additions:"+52"}
2025-01-28 11:10:48.420{changes:"0",file:"02_week_kestra/_files/macros/get_payment_type_description.sql",deletions:"-0",additions:"+17"}
2025-01-28 11:10:48.420{changes:"0",file:"02_week_kestra/_files/macros/macros_properties.yml",deletions:"-0",additions:"+12"}
2025-01-28 11:10:48.421{changes:"0",file:"02_week_kestra/_files/models/core/dim_zones.sql",deletions:"-0",additions:"+8"}
2025-01-28 11:10:48.422{changes:"0",file:"02_week_kestra/_files/models/core/fact_trips.sql",deletions:"-0",additions:"+56"}
2025-01-28 11:10:48.422{changes:"0",file:"02_week_kestra/_files/models/core/dm_monthly_zone_revenue.sql",deletions:"-0",additions:"+29"}
2025-01-28 11:10:48.423{changes:"0",file:"02_week_kestra/_files/models/core/schema.yml",deletions:"-0",additions:"+129"}
2025-01-28 11:10:48.424{changes:"0",file:"02_week_kestra/_files/models/staging/stg_green_tripdata.sql",deletions:"-0",additions:"+52"}
2025-01-28 11:10:48.424{changes:"0",file:"02_week_kestra/_files/models/staging/schema.yml",deletions:"-0",additions:"+196"}
2025-01-28 11:10:48.425{changes:"0",file:"02_week_kestra/_files/models/staging/stg_yellow_tripdata.sql",deletions:"-0",additions:"+48"}
2025-01-28 11:10:48.426{changes:"0",file:"02_week_kestra/_files/package-lock.yml",deletions:"-0",additions:"+6"}
2025-01-28 11:10:48.427{changes:"0",file:"02_week_kestra/_files/packages.yml",deletions:"-0",additions:"+5"}
2025-01-28 11:10:48.428{changes:"0",file:"02_week_kestra/_files/seeds/seeds_properties.yml",deletions:"-0",additions:"+9"}
2025-01-28 11:10:48.429{changes:"0",file:"02_week_kestra/_files/seeds/taxi_zone_lookup.csv",deletions:"-0",additions:"+266"}
2025-01-28 11:10:48.430{changes:"0",file:"02_week_kestra/docker-compose.yaml",deletions:"-0",additions:"+3"}
2025-01-28 11:10:48.430{changes:"0",file:"02_week_kestra/kestra-compose.yaml",deletions:"-0",additions:"+74"}
2025-01-28 11:10:48.431{changes:"0",file:"02_week_kestra/postgres-compose.yaml",deletions:"-0",additions:"+38"}
2025-01-28 11:10:48.432{changes:"0",file:"[README.md](http://readme.md/)",deletions:"-0",additions:"+6"}
```

**Comment by ramiz-rahman** [comment]
@anna-geller 

I finally managed to figure out the issue. One of my flows contained the GCS key (as shown in the tutorial videos) which caused Github to reject the push. However, Kestra did not show this error message (it was saying that the commit was a success), causing me to spend hours upon hours trying different permutations and combinations to isolate the source of the bug. 

If I had only received the correct error message (push failed due to the presence of secret keys), I would have solved it in a few. minutes. 

This will trip up a lot of users, especially new users like me, and I recommend this bug be treated with a higher priority.

**Comment by anna-geller** [comment]
TTBOMK we already provide a clear output in the Outputs tab indicating what has been pushed so this seems done - needs QA to validate

we could add a log msg identifying whether the push was successful, but it would be a duplicate of what we already provide in the output + if I'm not mistaken, we already provide that info on a TRACE log level -- also needs QA to validate

so far still the same next steps: **add error handling for when push doesn't work when commit signature verification is not enabled (BitBucket-specific)**

**Comment by ramiz-rahman** [comment]
Hey @anna-geller 

Thanks for jumping on this. 

So, that's the thing, the output and trace are not clear at all for this kind of failure. Or maybe I'm an idiot who can't read logs properly, but, hey that's still something to account for. 

Here's the Output of one of the failed calls. As you can see, a commit ID and commit URL are generated. This leads me to believe that the push succeeded. However, I went to GitHub and found that the push never came through. The generated URL is also incorrect and takes me to a 404 page. 

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/4cb728d5-c57d-441b-bc5d-4a8be755991b" />

Similarly, on the Logs tab, with the TRACE Filter enabled, there is nothing that indicates that the push failed. All I see is that the push has succeeded. 

<img width="1510" alt="Image" src="https://github.com/user-attachments/assets/19758798-cdf8-4dbd-bf16-92a941cbd91f" />

Let's contrast this with the case where the call failed because the current branch is up to date with contents inside Kestra. 

The call succeeds, but there is nothing to push. As expected, and the log message is crystal clear. 
<img width="1510" alt="Image" src="https://github.com/user-attachments/assets/acc75942-1008-4234-aa52-3ab80b048ecf" />

Similarly, inside the output tab, no commit message is generated. 

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/405ed5d6-8514-4d34-8a07-1e6f816ead23" />

--- 

In summary, the actual problem that I faced is: **in the case of a git push rejection**, the logs show that the **task succeeded** (it should show a failure or show success with a message that nothing was pushed) and in the output tab, it **generates a fake URL for the commit**

**Comment by anna-geller** [comment]
thx so much for the details and screenshots, very helpful!

**Comment by mgabelle** [comment]
The issue should be fixed by this PR : https://github.com/kestra-io/plugin-git/pull/127
From now on we check every push and if the status of the push is 'rejected' we'll throw an exception and the kestra task will fail.

@anna-geller This will be in the next release. I don't see the backport label so you can tell me if it's needed for old releases.
